pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/state/StateValue.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.functions.api.state;
+
+import java.nio.ByteBuffer;
+
+public class StateValue {
+    private final ByteBuffer value;
+    private final Long version;
+    private final Boolean isNumber;
+
+    public StateValue(ByteBuffer value, Long version, Boolean isNumber) {
+        this.value = value == null ? null : ByteBuffer.wrap(value.array());
```

Comment:
The javadoc of `ByteBuffer.array` method says that the method should only be called if `hasArray()` returns true.
The underlying array might have a specific position and limit. I think that `ByteBuffer.wrap(value.array())` is not a very good approach for duplicating a ByteBuffer.

Reply:
updated using `byte[]` instead of `ByteBuffer`